File: CODIGO/AdoptaYA-WebAPI/src/models/usuario.js
// src/models/usuario.js
const { DataTypes } = require('sequelize');

module.exports = (sequelize) => {
  const usuario = sequelize.define('usuario', {
    id:        { type: DataTypes.INTEGER.UNSIGNED, primaryKey: true, autoIncrement: true },
    username:  { type: DataTypes.STRING(25) },
    correo:    { type: DataTypes.STRING(100), allowNull: false, unique: 'ux_usuario_correo' },
    password:  { type: DataTypes.STRING(100), allowNull: false },
    nombre:    { type: DataTypes.STRING(100) },
    apellido:  { type: DataTypes.STRING(100) },
    date:      { type: DataTypes.DATE, allowNull: true },
    inactive:  { type: DataTypes.TINYINT(1) }
  }, {
    tableName: 'usuario',
    freezeTableName: true,
    timestamps: false
  });

  usuario.associate = (models) => {
    usuario.belongsToMany(models.rol, {
      through: models.rol_usuario,
      foreignKey: 'id_usuario',
      otherKey: 'id_rol'
    });
  };

  return usuario;
};

File: DOCUMENTOS/CODIGO/AdoptaYA/Functionalities/Permissions/Model/Usuario.cs
using System.Text.Json.Serialization;

namespace AdoptaYA.Functionalities.Permissions.Model;
public class Usuario
{
    [JsonPropertyName("id")]
    public int Id { get; set; }

    [JsonPropertyName("username")]
    public string? Username { get; set; }

    [JsonPropertyName("correo")]
    public string? Email { get; set; }

    [JsonPropertyName("password")]
    public string? Password { get; set; }

    [JsonPropertyName("nombre")]
    public string? FirstName { get; set; }

    [JsonPropertyName("apellido")]
    public string? LastName { get; set; }

    [JsonPropertyName("date")]
    public DateTime? Date { get; set; }

    [JsonPropertyName("inactive")]
    public int Inactive { get; set; }
}